esx_service/cli/vmdkops_admin.py

```diff
@@ -726,38 +731,382 @@ def get_version():
+    """ Handle tenant vm add command """
+    error_info, tenant = get_tenant_from_db(args.name)
+    if error_info:
+        return operation_fail(error_info)
+    
+    if not tenant:
+        error_info = "Tenant {0} does not exist".format(args.name)
+        return operation_fail(error_info)
+
+    error_info, vms = generate_tuple_from_vm_list(args.vm_list)
+    if error_info:
+        return operation_fail(error_info)
+ 
+    error_info = tenant.add_vms(auth._auth_mgr.conn, vms)
+
+    if error_info:
+        return operation_fail(error_info)
+    else:
+        print "tenant vm add succeeded"
+  
 def tenant_vm_rm(args):
-    print(NOT_IMPLEMENTED)
+    """ Handle tenant vm rm command """
+    error_info, tenant = get_tenant_from_db(args.name)
+    if error_info:
+        return operation_fail(error_info)
+    
+    if not tenant:
+        error_info = "Tenant {0} does not exist".format(args.name)
+        return operation_fail(error_info)
+
+    error_info, vms = generate_tuple_from_vm_list(args.vm_list)
+    
+    if error_info:
+        return operation_fail(error_info)
+      
+    error_info = tenant.remove_vms(auth._auth_mgr.conn, vms)
+
+    if error_info:
+        return operation_fail(error_info)
+    else:
+        print "tenant vm rm succeeded"
+
+def tenant_vm_ls_headers():
+    """ Return column names for tenant vm ls command """
+    headers = ['Uuid', 'Name']
+    return headers
+
+def generate_tenant_vm_ls_rows(vms):
+    """ Generate output for tenant vm ls command """
+    rows = []
+    for vm in vms:
+        # vm has the format like this (vm_uuid, vm_name)
+        uuid = vm[0]
+        name = vm[1]
+        rows.append([uuid, name])
+    
+    return rows
+
 
 def tenant_vm_ls(args):
-    print(NOT_IMPLEMENTED)
+    """ Handle tenant vm ls command """
+    error_info, tenant = get_tenant_from_db(args.name)
+    if error_info:
+        return operation_fail(error_info)
+    
+    if not tenant:
+        error_info = "Tenant {0} does not exist".format(args.name)
+        return operation_fail(error_info)
+   
+    header = tenant_vm_ls_headers()
+    rows = generate_tenant_vm_ls_rows(tenant.vms)
+    print(cli_table.create(header, rows))
+
+def default_privileges():
+     privileges = [{'datastore': 'datastore1',
+                     'global_visibility': 0,
+                      'create_volume': 0,
+                      'delete_volume': 0,
+                     'mount_volume': 0,
+                      'max_volume_size': 0,
+                      'usage_quota': 0}]
+     return privileges
+
+def set_privileges(rights, privileges, value):
+    """ set or unset privileges based rights passed by command line """
+    if 'create' in rights:
+        privileges[auth_data_const.COL_CREATE_VOLUME] = value
+
+    if 'delete' in rights:
+        privileges[auth_data_const.COL_DELETE_VOLUME] = value
+
+    if 'mount' in rights:
+        privileges[auth_data_const.COL_MOUNT_VOLUME] = value
+
+    if 'all' in rights:
```

Comment:
Instead check for 'all' privileges first and set the privileges and exit the function vs. checking each privilege and finally setting all privileges.